src/ocean/io/serialize/StructSerializer.d

```diff
@@ -575,12 +575,8 @@ struct StructSerializer ( bool AllowUnions = false )
 
                 static if (isTypedef!(T))
                 {
-                    mixin(`
-                    static if ( is(T B == typedef) )
-                    {
-                        serializer.serialize(data, cast(B)(field), field_name);
-                    }
-                    `);
+                    serializer.serialize(data, cast(StripTypedef!(T))(field),
```

Comment:
Yes, but intended usage of this feature is not limited to legacy `Typedef`. As I have mentioned previously, after we drop D1 support in ocean I want to modify `Contiguous` to say `ref S value(); alias value this; alias BaseType = S;` and get it processed by `Formatter` like `S` for free. There are many more places where one may want well-defined subtyping relation but if your utilities operate in typedef terms, it becomes applicable only for typedef-like entities.